src/mcts/search.cc

```diff
@@ -1145,6 +1145,22 @@ SearchWorker::NodeToProcess SearchWorker::PickNodeToExtend(
       }
       return NodeToProcess::Collision(node, depth, collision_limit);
     }
+    // Probably best place to check for two-fold draws consistently.
+    // Depth starts with 1 at root, so real depth is depth - 1
+    if (node->IsTwofoldTerminal() && ( params_.GetTwoFoldDrawLevel() > 0 ) ) {
+      LOGFILE << "== encountered twofold draw == depth: " << depth - 1;
+      if (params_.GetTwoFoldDrawLevel() == 2 && depth - 1 <= 3) {
+        // Level 2: no two-fold draw at depth 3 or lower
+        LOGFILE << "== level 2 twofold draw reset ==";
+        node->MakeNotTerminal();
+      } else if (params_.GetTwoFoldDrawLevel() == 1 && depth - 1 < node->GetM()) {
+        // Level 1: check whether first repetition was before root
+        // Length of repetition was stored in m_, so we don't need to calculate
+        LOGFILE << "== level 1 twofold draw reset ==";
+        node->MakeNotTerminal();
```

Comment:
You can also combine the multiple branch bodies with a single condition as in the other code path. Also Tilps' comment about `MakeNotTerminal` not fixing up ancestors still applies as this happens for nodes not at root now: https://github.com/LeelaChessZero/lc0/pull/1197#discussion_r436212969

Reply:
Combined the statements into a single one.